packages/base-driver/lib/basedriver/helpers.js

```diff
@@ -18,13 +18,14 @@ const CACHED_APPS_MAX_AGE = 1000 * 60 * 60 * 24; // ms
 const APPLICATIONS_CACHE = new LRU({
   maxAge: CACHED_APPS_MAX_AGE, // expire after 24 hours
   updateAgeOnGet: true,
-  dispose: async (app, {fullPath}) => {
-    if (!await fs.exists(fullPath)) {
-      return;
-    }
-
-    logger.info(`The application '${app}' cached at '${fullPath}' has expired`);
-    await fs.rimraf(fullPath);
+  dispose: (app, {fullPath}) => {
+    logger.info(`The application '${app}' cached at '${fullPath}' has ` +
+      `expired after ${CACHED_APPS_MAX_AGE}ms`);
+    setTimeout(async () => {
```

Comment:
what is this for?

Reply:
to run the async snippet

Reply:
no, I mean, why is `setTimeout` here?

Reply:
rimraf is safe to call, it should not create any exceptions

Reply:
i generally don't like the `setTimeout` method of calling async functions in a non-async context. I'd rather do something like:

```js
(async () => { /* async function logic */ })().catch(log.error.bind(log));
```

or to make it more readable, define the function and call it in 2 steps

```js
async function foo () {
 /* ... */
}
foo().catch(log.error)
```

Adding `.catch` is just a safeguard against unhandled rejections. By running async code in a non async context we are already agreeing that we don't care if something bad happens, but we might as well log it just in case something does happen, however unlikely.